compiler/src/dotty/tools/dotc/core/SymUtils.scala

```diff
@@ -249,6 +249,10 @@ class SymUtils:
     def caseAccessors(using Context): List[Symbol] =
       self.info.decls.filter(_.is(CaseAccessor))
 
+    // TODO: Check if `Synthetic` is stamped properly
+    def javaRecordComponents(using Context): List[Symbol] =
+      self.info.decls.filter(sym => sym.is(Synthetic) && sym.is(Method) && !sym.isConstructor)
```

Comment:
Check if this works if the user overrides in both phases, otherwise we'll need to stamp an extra flag on the methods.